scanner/src/test/kotlin/utils/FileListingResolverTest.kt

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (C) 2023 The ORT Project Authors (see <https://github.com/oss-review-toolkit/ort/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.scanner.utils
+
+import io.kotest.core.spec.Spec
+import io.kotest.core.spec.style.StringSpec
+import io.kotest.matchers.shouldBe
+
+import java.io.File
+
+import org.ossreviewtoolkit.model.ArtifactProvenance
+import org.ossreviewtoolkit.model.RemoteArtifact
+import org.ossreviewtoolkit.model.toYaml
+import org.ossreviewtoolkit.model.utils.FileProvenanceFileStorage
+import org.ossreviewtoolkit.utils.ort.storage.LocalFileStorage
+import org.ossreviewtoolkit.utils.test.createTestTempDir
+
+class FileListingResolverTest : StringSpec({
+    "resolve() should create the expected file listing" {
+        val resolver = FileListingResolver(
+            storage = FileProvenanceFileStorage(
+                storage = LocalFileStorage(createTestTempDir()),
+                filename = "bytes",
+            ),
+            provenanceDownloader = {
+                createTestTempDirWithFiles(
+                    ".git/index",
+                    "LICENSE",
+                    "src/cli/main.cpp"
+                )
+            },
+        )
+
+        val fileListing = resolver.resolve(ArtifactProvenance(sourceArtifact = RemoteArtifact.EMPTY))
+
+        fileListing.toYaml() shouldBe File("src/test/assets/expected-file-listing.yml").readText()
```

Comment:
It would be good to have some more fine grained test, e.g. to verify that the exclude patterns work as expected. I'm fine with adding them later.
If more tests are added, it might be better for the overview to inline the expected results in the tests instead of using separate files.

Reply:
I'll do it in a follow up PR then.